### ISSUE TITLE
Implement key events for tracking in Matomo

### ISSUE THREAD

**Issue by talmahaj** [tari-project/wallet-android#271]
@talmahaj commented on [Wed Apr 08 2020](https://github.com/tari-project/wallet-ios/issues/261)

See the events we can trigger via mobile at https://github.com/tari-project/wallet-android/wiki/Matomo-Event-Tracking

**Comment by Tas4tari** [comment]
@kukabi How do i ensure that the events as listed are successfully tracked?

**Comment by Tas4tari** [comment]
Testing via Huawei P30 lite:
Please see the video below:
https://www.loom.com/share/4de0408c567943a8886ba98e31cc64a9

**Comment by Tas4tari** [comment]
Transaction accepted synchronous:
https://www.loom.com/share/5f536c5b360841d09b077e0399d0641c

**Comment by Tas4tari** [comment]
Transaction failed: Node issue:
https://www.loom.com/share/c8d85ee5d8464d03b601cfdf530abeae

**Comment by talmahaj** [comment]
@kukabi did we determine if there is any possible way to exclude event data from test build users so it doesn't impact reporting, since we can't do the debug mode thing we're doing in iOS?

**Comment by talmahaj** [comment]
@kukabi I'm seeing a discrepancy between the #s of the /finalize page view and the transaction initiated event that I wouldn't expect -- take a look and let me know what you think?

**Comment by kukabi** [comment]
> @kukabi did we determine if there is any possible way to exclude event data from test build users so it doesn't impact reporting, since we can't do the debug mode thing we're doing in iOS?

Unfortunately not - only way could be to have a checkbox in the debug screen that turns off tracking. Otherwise, the Android APIs provide no info about whether the app is currently in production or not.

**Comment by kukabi** [comment]
> @kukabi I'm seeing a discrepancy between the #s of the /finalize page view and the transaction initiated event that I wouldn't expect -- take a look and let me know what you think?

Sure, will look 👍  @talmahaj

**Comment by kukabi** [comment]
@talmahaj I just had a look at the data from 11th to 19th of May, it looks like this:

Date | Transaction Initiated | Send Tari - Finalize
:---- | --------------------: | -------------------:
19.05.2020 | 36 | 36
18.05.2020 | 122 | 122
17.05.2020 | 26 | 27
16.05.2020 | 29 | 30
15.05.2020 | 60 | 59
14.05.2020 | 65 | 65
13.05.2020 | 37 | 37
12.05.2020 | 70 | 69
11.05.2020 | 50 | 50

They seem to differ by 1 max on these nine days.

The difference on the days of the week of the 4th is more drastic, I don't have an idea why, looking at the code I'm sure the page view (`Send Tari - Finalize`) cannot ever happen without the event (`Transaction Initiated`).

I'm moving this issue to UAT for the moment. Let's observe whether the discrepancy increases over time.

**Comment by kukabi** [comment]
Not sure about this, but the discrepancy on the week of the 4th may also be related to #366. @nyarian

**Comment by nyarian** [comment]
@kukabi that's possible. If a configuration is changed when on `FinalizeSendTxFragment` then all the finalization process is restarted and log is issued once again.

**Comment by Tas4tari** [comment]
@kukabi @talmahaj @nyarian I still don't see the Transaction failed - Tor issue in the event
Can you please guide me on how I view the Transaction initiated VS Send Tari - Finalize table

<img width="954" alt="Screen Shot 2020-05-26 at 11 10 46 AM" src="https://user-images.githubusercontent.com/50987039/82882588-8e75d900-9f41-11ea-9b6a-518b67202ad8.png">

**Comment by Tas4tari** [comment]
Investigate Transaction - stored in Transaction Accepted - Synchronous

**Comment by Anziskavds** [comment]
I'm currently testing the Testflight build version 0.2.0. I will test this as soon as I installed the Prod app again.

cc @kukabi @talmahaj

**Comment by Anziskavds** [comment]
Tested successfully on Android version 0.2.1 on a Samsung S9.

The event tracking names is the same on both platforms(iOS and Android) but the names aren't the same as per the requirements. I have confirmed with Tal if she's happy with it and said it looks great.
Document with event names: https://docs.google.com/document/d/1r0uy5fDa-coD55vKLwzUL0bAP008fPQmiBHIV0qF_rY/edit?usp=sharing

Test Results screenshots:

https://share.getcloudapp.com/12uNkZKP

https://share.getcloudapp.com/Wnub6PBR

https://share.getcloudapp.com/p9uPZm86

cc @talmahaj @Tas4tari
